internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/AtuwaDownloader.java

```diff
@@ -0,0 +1,103 @@
+ *  under the License.
+ * /
+ */
+
+package org.wso2.security.tools.scanner.dependency.js.preprocessor;
+
+import org.apache.log4j.Logger;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.DownloaderException;
+import org.wso2.security.tools.scanner.dependency.js.model.Product;
+import org.wso2.security.tools.scanner.dependency.js.utils.HttpDownloadUtility;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Download product pack from Atuwa.
+ * This class extends ResourceDownloader abstract class and implements downloadProductPack method.
+ */
+public class AtuwaDownloader extends ResourceDownloader {
+
+    private static final Logger log = Logger.getLogger(PreProcessor.class);
+
+    @Override
+    public List<String> downloadProductPack(Product productDto, String path) {
+        return download(path);
+    }
+
+    private List<String> download(String path) {
+        BufferedReader reader = null;
+        List<String> zipFilePathList = null;
+        String name = null;
+        try {
+            zipFilePathList = new ArrayList<>();
+            String urlString = JSScannerConstants.ATUWA_BASE_URL + JSScannerConstants.OB_INDEX_FILE;
+            String downloadUrl;
+            // create the url
+            URL url = new URL(urlString);
+
+            // open the url stream, wrap it an a few "readers"
+            reader = new BufferedReader(new InputStreamReader(url.openStream()));
+
+            // write the output to stdout
+            String line;
+            File tarDir = null;
+            while ((line = reader.readLine()) != null) {
+                String element[] = line.split(":");
+                Long dateDiff = getDateDiffFromLastWeeklyRelease(element[1]);
+                if (dateDiff < 8) {
+                    if (isWeeklyRelease(element[0])) {
+                        tarDir = new File(path + File.separator + "weeklyRelease");
+                    } else if (isGARelease(element[0])) {
+                        tarDir = new File(path + File.separator + "GARelease");
+                    }
+                    if (tarDir != null) {
+                        createDirectory(tarDir);
+                    }
+                    name = element[0];
+                    downloadUrl = JSScannerConstants.ATUWA_BASE_URL + element[0];
+                    if (path != null) {
+                        assert tarDir != null;
+                        zipFilePathList.add(HttpDownloadUtility.downloadFile(downloadUrl, tarDir.getAbsolutePath()));
+                    }
+                } else {
+                    break;
+                }
+            }
+        } catch (ParseException | DownloaderException | IOException e) {
```

Comment:
catch the exceptions where it occurs, rather than catching at the end.

Reply:
Resolved